### PR TITLE
rectpacking: Add padding to area calculation.

### PR DESCRIPTION
**Pull request by soerendomroes**
Rework the rectpacking algorithm to take padding into account. This reworks most of the algorithms spacing calculation since previously all blocks and rows always added the spacing to their height, which cases problems with the aspect ratio calculation.
(Also fixed some bugs as a result of this).

- [x] test case: stack is put next to stack with multiple blocks
- [x] test case: too many blocks to absorb into the current row
- [x] test SCChart region collision and expansion (seems to be broken for height)

Signed-off-by: Soeren Domroes <sdo@informatik.uni-kiel.de>

### CODE REVIEW

**Comment by soerendomroes** [comment]
More tests should be added for rectpacking compaction step:

- [ ] stack is put next to stack with multiple blocks

File: plugins/org.eclipse.elk.alg.rectpacking/src/org/eclipse/elk/alg/rectpacking/util/DrawingData.java [hunk 1]
```diff
@@ -196,4 +198,11 @@ public double getNextYcoordinate() {
     public void setNextYcoordinate(final double potentialY) {
         this.nextYcoordinate = potentialY;
     }
+
+    /**
+     * @return the dar
+     */
+    public double getDar() {
```
Comment (uruuru) [inline plugins/org.eclipse.elk.alg.rectpacking/src/org/eclipse/elk/alg/rectpacking/util/DrawingData.java:205]: I'd suggest to call it `getDesiredAspectRatio()`.

**Review by uruuru — COMMENTED** [review]
I'm not familiar enough with the code to properly judge the changes. So unless @le-cds objects feel free to merge :). 

One thing that caught my eye: did you use autoformatting? Some lines and comments look somewhat off.

**Comment by soerendomroes** [comment]
> One thing that caught my eye: did you use autoformatting? Some lines and comments look somewhat off.

Do you have an example for this?

**Comment by soerendomroes** [comment]
@le-cds If you have nothing to add I would like to merge this.

**Pull request merged**